### PR TITLE
rework title processing

### PR DESCRIPTION
**Pull request by osoukup**
This PR has three parts. The main is the rework of the OSIDB title processing with respect to the Bugzilla summary mapping. The EMBARGOED and CVE IDs parts are just stripped (embargo status is determined by Bugzilla groups and CVE IDs are stored in Bugzilla aliases) and the component and the title are parsed from the remaining. This includes introducing the component as the full-blown attribute. Also the opposite direction from OSIDB to Bugzilla is covered.

Additionally I resurrected a few long-time commented out tests as I wanted to use `test_fixups.py` file.

Closes OSIDB-694

### CODE REVIEW

**Review by costaconrado — APPROVED** [review]
LGTM.

**Review by JakubFrejlach — APPROVED** [review]
Couple of suggestions, otherwise LGTM.

File: osidb/tests/factories.py [hunk 1]
```diff
@@ -50,6 +50,7 @@ class Meta:
     impact = factory.Faker(
         "random_element", elements=list(set(FlawImpact) - {FlawImpact.NOVALUE})
     )
+    component = factory.fuzzy.FuzzyText()
```
[the remark this answers is not in the document]
Comment (osoukup) [reply osidb/tests/factories.py:53]: Thank you for the advice! I will replace it

File: osidb/tests/test_endpoints.py [hunk 2]
```diff
@@ -1069,6 +1070,7 @@ def test_flaw_create(self, auth_client, test_api_uri):
         response = auth_client.get(f"{test_api_uri}/flaws/{created_uuid}")
         assert response.status_code == 200
         assert response.json()["cve_id"] == "CVE-2021-0666"
+        assert response.json()["component"] == "curl"
```
[the remark this answers is not in the document]
Comment (osoukup) [reply osidb/tests/test_endpoints.py:1073]: This is an integration test verifying that we can set the component via API and also retrieve it via API. It should ideally be done for all the attributes but this PR specifically deals with `component`.

File: apps/bbsync/tests/test_integration.py [hunk 3]
```diff
@@ -60,6 +60,8 @@ def test_flaw_create(self, auth_client, test_api_uri):
         response = auth_client.get(f"{test_api_uri}/flaws/{created_uuid}")
         assert response.status_code == 200
         assert response.json()["cve_id"] == "CVE-2021-0773"
+        assert response.json()["title"] == "Foo"
+        assert response.json()["component"] == "curl"
```
[the remark this answers is not in the document]
Comment (osoukup) [reply apps/bbsync/tests/test_integration.py:64]: This is an integration test verifying that the `BBSync` composes the Bugzilla summary well based on the title, component, and CVE IDs and then that `bzimport` parses it well back again and the resulting title is only a summary part after the colon and the component is correctly fetched from there. I think this is the only integration test testing this all together.

File: apps/bbsync/tests/test_query.py [hunk 4]
```diff
@@ -20,6 +20,130 @@
...
+                "space",
+                "CVE-1000-12345",
+                "CVE-2000-12345",
+                True,
+                "cucumber",
+                "EMBARGOED CVE-1000-12345 CVE-2000-12345 space: cucumber",
+            ),
+        ],
+    )
+    def test_generate_summary_multi_cve(
```
[the remark this answers is not in the document]
Comment (osoukup) [reply apps/bbsync/tests/test_query.py:124]: You are correct that some parts are never changing. I will think about simplifying it.

File: osidb/tests/test_endpoints.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (osoukup) [reply osidb/tests/test_endpoints.py:1073]: The main difference between this and the other integration test you are asking about is that here the two-way sync if turned off while it is turned on at the other case. It significantly influence the `save` behavior so IMO makes it worth testing both cases.

File: apps/bbsync/tests/test_integration.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (JakubFrejlach) [reply apps/bbsync/tests/test_integration.py:65]: Got it.

**Pull request merged**